### PR TITLE
fix asdf schema examples

### PR DESCRIPTION
**Pull request by braingram**
Downstream testing in asdf revealed a few issues with the examples included in a few sunpy schemas.
https://github.com/asdf-format/asdf/actions/runs/6828518321/job/18572820175?pr=1677

This PR fixes the examples (formatting them so the asdf pytest plugin can load and test the examples).

The issue appears to be that the tests are running against the installed version of sunpy:
```
[gw0] [  0%] PASSED ../../.tox/py311/lib/python3.11/site-packages/sunpy/coordinates
```
whereas the configuration for the asdf schemas test assumes testing against the checked out code:
https://github.com/sunpy/sunpy/blob/cfb7ed4c5b0411cb157a3cea6d56220e5c7b126b/setup.cfg#L162

I don't see a way to configure the current sunpy ci to allow the asdf pytest plugin to run the tests without adding a job that doesn't using `--pyargs` and tests against the checked out code. This is a limitation of the asdf pytest plugin and should be addressed there likely even if a workaround can be found.

I'm going to open this PR for review with the knowledge that the CI is not testing the asdf schemas so that hopefully the schema issues can be addressed and a later PR can update the CI to test the schemas.

### CODE REVIEW

File: sunpy/io/special/asdf/resources/schemas/solarmagnetic-1.0.0.yaml [hunk 1]
```diff
@@ -13,7 +13,9 @@ examples:
         !<tag:sunpy.org:sunpy/coordinates/frames/solarmagnetic-1.0.0>
           frame_attributes:
             magnetic_model: igrf13
-
+  -
+    - A Solar Magnetic frame with and obstime
```
Comment (ayshih) [inline sunpy/io/special/asdf/resources/schemas/solarmagnetic-1.0.0.yaml:17]:
```suggestion
    - A Solar Magnetic frame with an obstime
```

**Review by ayshih — APPROVED** [review]
Fixed a typo

**Review by nabobalis — APPROVED** [review]

**Comment by ayshih** [comment]
Gah, you merged before I pulled off the 5.0 backport label!

**Comment by ayshih** [comment]
(I messed up these schemas I added in #7239, which is only in 5.1)

**Comment by nabobalis** [comment]
Sorry!

**Pull request merged**